### PR TITLE
Fix GPS Rescue parameters confusion

### PR DESCRIPTION
**Pull request by haslinghuis**
- Fix confusing parameter naming
- Add initialClimb parameter
- Add course over ground to heading
- Fix range for some parameters

Firmware PR: https://github.com/betaflight/betaflight/pull/13047

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by Asizon — APPROVED** [review]
Looks good!

**Comment by ctzsnooze** [comment]
Thanks @haslinghuis for these changes!

This is a screen dump showing both pyGPSclient and Betaflight Configurator handling the same data coming from a GPS in parallel.

Note that the GPS heading is shown in degrees, and matches the 'track' value in pyGPSclient.  All other values match up also.

Screen Shot 2023-08-24 at 09 31 21

**Comment by ctzsnooze** [comment]
When I go to check the Failsafe tab, the labels for the GPS rescue settings are now correct, and the message warning that return altitude only applies in FIXED altitude mode is very clear.

I tested using the Betaflight PR #13047, and it seems are some issues to fix (not sure which end):
- The `gps_rescue_initial_climb` field is blank, it doesn't appear to read the value from the CLI
- If I adjust `gps_rescue_initial_climb` to a value, and save, the value is not saved
- The `gps_rescue_return_alt`  range is 2-255m, default of 30m.  In Configurator, with this PR, the range is 20-100m.  Now frankly 20m return height is probably a safer minimum than 2m, but the user should be able to configure it down to say 5m, which should still be safe.  Most likely we should set minimum to 5m in both places.  The maximum height could be whatever the user wanted, within reason, and the two maximum values should match.  Looks like we now have 16 bits, the old maximum of 255 was limited by 8 bits.  In some mountain surfing situations I guess the user could want to set the height to 1000m; perhaps then the range should be 5-1000m in both firmware and configurator.


Screen Shot 2023-08-24 at 09 43 36

**Comment by ctzsnooze** [comment]
One minor detail that I noticed, in relation to our 3D fix.

In Configurator we have an icon next to the text '3D fix' on the right.
I noticed that pyGPSclient reports a 3D fix sometimes well *before* we light up our 3D fix icon in green.
To get a 3D fix requires, for most modules, a minimum of 4 sats.
A 2D fix can be achieved with just 3 sats.

Here is an image showing a 2D fix with an erroneous altitude value, early on in a connection phase, with pyGPSclient reporting on the raw GPS values at the top:
Screen Shot 2023-08-24 at 10 14 34
Note that Betaflight shows the GPS icon, at the top, as being Red still.

Here is the situation where we get our 3D fix, with 5 sats in the solution.  The altitude value is a bit different now.
Screen Shot 2023-08-24 at 10 17 57

A bit later on we get 3D fix with 11 sats in the solution.  Horizontal accuracy (hAcc) is 1.29m and vertical accuracy (vAcc) is 2.27m.  This is quite a lot better than before, but there isn't a clear graphical indication that we now have the 8 sats required for us to set a home point.

Screen Shot 2023-08-24 at 10 21 20

I wonder if we could change the display of the satellite number on the right so that it is shown like the 3D fix icon, ie like an icon or button, and make it:
- red when there are not enough sats to have a 3D fix
- orange when we have a 3D fix but less sats than the value of `gps_rescue_min_sats` 
- green when we have a 3D fix and more sats than `gps_rescue_min_sats` (ie, 8 or more sats, by default)

I wonder also if we could display the map as soon as we get any coordinates at all, rather than waiting for the 3D fix.  Currently no map is shown until we get a 3D fix.

But it might be quite nice to see the map as soon as we receive latitude and longitude values.  My reasoning is that the position indicated on the map would then run around all over the place initially, and as the number of sats increased, so would its accuracy.  This might help people who choose the 'arm without home point' option visualise just how dangerous this is.

File: src/tabs/failsafe.html
Comment (haslinghuis) [inline src/tabs/failsafe.html:127]: this will require some semver magic :)

Comment (haslinghuis) [reply src/tabs/failsafe.html]: Done

**Comment by haslinghuis** [comment]
> The gps_rescue_initial_climb field is blank, it doesn't appear to read the value from the CLI

Sorted now

**Comment by ctzsnooze** [comment]
Hmm, no, it seems like the gps_rescue_initial_climb field is still blank, and doesn't appear to read the value from the CLI.  If I up-arrow it then shows '5', even if the value is 7 in the CLI.

**Comment by ctzsnooze** [comment]
Please change the html for theGPS rescue parameters that currently default to two decimal places, so that they default to 1  decimal place, and adjust in 0.1 increments (not 0.01).  Thanks.

File: src/tabs/failsafe.html
Comment (ctzsnooze) [inline src/tabs/failsafe.html:150]: Please change step here to 0.1, and only once decimal place for min and max, eg:
`min="3.0" max="30.0" step="0.1"`

File: src/js/tabs/failsafe.js
Comment (ctzsnooze) [inline src/js/tabs/failsafe.js:312]: Please refactor this code in a simpler but functionally equivalent form, so that we check each version sequentially.
This makes it much easier to review and manage the settings.
The following code provides correct values for 4.4 and 4.5, and I strongly recommend using it:
```
        if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) {
            $('input[name="gps_rescue_min_dth"]').attr("min", 50);
            $('input[name="gps_rescue_min_dth"]').val(FC.GPS_RESCUE.minRescueDth);
        } else {
            $('input[name="gps_rescue_min_dth"]').closest('.number').hide();
        }

        if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_45)) {
            $('input[name="gps_rescue_min_dth"]').attr("min", 20);
            $('input[name="gps_rescue_ascend_rate"]').attr("min", 0.5);
            $('input[name="gps_rescue_ascend_rate"]').attr("max", 25);
            $('input[name="gps_rescue_descent_distance"]').attr("min", 5);
            $('input[name="gps_rescue_descend_rate"]').attr("min", 0.25);
            $('input[name="gps_rescue_return_altitude"]').attr("min", 2);
            $('input[name="gps_rescue_return_altitude"]').attr("max", 255);
            $('input[name="gps_rescue_ground_speed"]').attr("min", 0);
            $('input[name="gps_rescue_ground_speed"]').attr("max", 30);
            $('input[name="gps_rescue_angle"]').attr("max", 80);
        }

        if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_46)) {
            $('input[name="gps_rescue_min_dth"]').attr("min", 10);
            $('input[name="gps_rescue_min_dth"]').attr("max", 30);
            $('input[name="gps_rescue_return_altitude"]').attr("min", 5);
            $('input[name="gps_rescue_return_altitude"]').attr("max", 1000);
            $('input[name="gps_rescue_angle"]').attr("min", 30);
            $('input[name="gps_rescue_angle"]').attr("max", 60);
            $('input[name="gps_rescue_descent_distance"]').attr("min", 10);
            $('input[name="gps_rescue_initial_climb"]').attr("min", 0);
            $('input[name="gps_rescue_initial_climb"]').attr("max", 100);
            $('input[name="gps_rescue_initial_climb"]').val(FC.GPS_RESCUE.initialClimbM);
        } else {
            $('input[name="gps_rescue_initial_climb"]').closest('.number').hide();
        }
```

File: src/js/tabs/failsafe.js
Comment (ctzsnooze) [inline src/js/tabs/failsafe.js:302]: Please just change the step resolution in the html, so that all users benefit, regardless of their firmware level.
The previous 0.01m/s resolution for velocity settings was just crazy.

File: src/tabs/failsafe.html
Comment (ctzsnooze) [inline src/tabs/failsafe.html:143]: These are not the correct values for `gps_rescue_initial_climb`.  They should be min="0" max="100".
It's the `gps_rescue_return_altitude` that has the range 5 to 1000.

File: src/js/tabs/failsafe.js
Comment (ctzsnooze) [reply src/js/tabs/failsafe.js:302]: Update - to see a value in Configurator without two decimal points, we need to set the input `toFixed(1)`, not `toFixed(2)`.
I recommend we do this, and change the html to use 0.1 steps, and change the default numbers in html to only one decimal place.
Being consistent throughout makes the code simpler.
And it won't adversely affect older versions, I expect that people using older versions will prefer to only have once decimal point to worry about.
Just changing the step in the javascript to 0.1 isn't enough.

**Review by ctzsnooze — APPROVED** [review]

**Pull request merged**
